gitlab.go

```diff
@@ -291,8 +294,7 @@ func (c *Client) Do(req *http.Request, v interface{}) (*Response, error) {
 
 	response := newResponse(resp)
 
-	err = CheckResponse(resp)
-	if err != nil {
+	if err := CheckResponse(resp); err != nil {
```

Comment:
Are you saying that the new code will be slower because it uses an extra variable via shadowing? I highly doubt that's the case :)

This is the preferred style. On one hand, shadowing instead of reassigning. This way you can move the line around without worrying if there was a `var err error` already declared.

On the other hand, a separate `if err != nil {` only makes sense if the function returned more than just an error.